Engine/Setup/BrokerageSetupHandler.cs

```diff
@@ -544,7 +546,11 @@ private void AddUnrequestedSecurity(IAlgorithm algorithm, Symbol symbol)
                     continue;
                 }
 
-                AddUnrequestedSecurity(algorithm, order.Symbol);
+                var security = AddUnrequestedSecurity(algorithm, order.Symbol);
+                if (security != null)
+                {
+                    order.PriceCurrency = security.SymbolProperties.QuoteCurrency;
```

Comment:
If the security was already added `AddUnrequestedSecurity` will return `null` and the price currency will not be set, but it should be.
I think it would be nice if this `if` wasn't required, `AddUnrequestedSecurity` could return the security always instead, maybe using `algorithm.Securities.TryGetValue()`?


Edit: actually even better: `order.PriceCurrency` is already being set by the BTH when creating a new order, so it would be nice to make him also set it for open orders, through `transactionHandler.AddOpenOrder`. Moving `AddUnrequestedSecurity` above the call to `AddOpenOrder` will allow the BTH to fetch the security internally like it does when `HandleSubmitOrderRequest`